site/testsuites/testsuite-kotlin/src/test/kotlin/docs/tbdex/wallet/ManageCredentialsTest.kt

```diff
@@ -0,0 +1,137 @@
+        val offering = mockOfferings.first()
+
+        // :snippet-start: retrievePresentationDefinitionFromOfferingsRequiredClaimsKt
+        val presentationDefinition = offering.data.requiredClaims 
+        //  :snippet-end:
+        ?: fail("Presentation definition should not be null")
+
+        assertEquals("custom-pd-id", presentationDefinition.id, "Presentation definition ID mismatch")
+        assertEquals(2, presentationDefinition.inputDescriptors.size, "Input descriptors size mismatch")
+    }
+
+    @Test
+    fun `match offerings with specific criteria`() {
+        val pfiDids = listOf(pfiDid)
+        // :snippet-start: findMatchingOfferingsWithCredentialValidationKt
+        val payinCurrencyCode = "USD" // Desired payin currency code
+        val payoutCurrencyCode = "KES" // Desired payout currency code
+
+        // Customer's signed credentials in JWT format
+        val credentials = listOf(vcJwtResidence, vcJwtSanctions) 
+
+        // Array to store the matched offerings
+        val matchedOfferings = ArrayList<Offering>()
+
+        for (pfiDid in pfiDids) {
+            val offerings = TbdexHttpClient.getOfferings(pfiDid)
+            
+            // Filter offerings based on the desired currency pair
+            offerings.filter { offering ->
+                offering.data.payinCurrency.currencyCode == payinCurrencyCode && offering.data.payoutCurrency.currencyCode == payoutCurrencyCode
+            }.forEach { offering ->
+
+                // Extract the presentation definition from the offering
+                offering.data.requiredClaims?.let { presentationDefinition ->
+                    try {
+                        //highlight-start
+                        // Validate customer's VCs against the offering's presentation definition
+                        PresentationExchange.satisfiesPresentationDefinition(
+                            credentials, presentationDefinition
+                        )
+                        //highlight-end 
+
+                        //  Add offerings that match the customer's needs and qualifications
+                        matchedOfferings.add(offering)
+                    } catch (e: Exception) {
+                        // Offerings where the customer's VCs do not meet the requirements are skipped
+                    }
+                }
+            }
+        }
+        // :snippet-end:
+        assertNotEquals(0, matchedOfferings.size, "No matching offerings found")
+    }
+
+    @Test
+    fun `selects credentials based on required claims from offering`() {
+        val credentials = listOf(vcJwtResidence, vcJwtSanctions)
+        val selectedOffering = mockOfferings.get(0)
+        // :snippet-start: getSelectedCredentialsKt
+        // Select the credentials to be used for the exchange
+        val selectedCredentials = PresentationExchange.selectCredentials(credentials, selectedOffering.data.requiredClaims!!)
```

Comment:
```suggestion
        val selectedCredentials = PresentationExchange.selectCredentials(
            credentials, 
            selectedOffering.data.requiredClaims!!
        )
```

was getting cut off

<img width="854" alt="image" src="https://github.com/TBD54566975/developer.tbd.website/assets/15972783/0aeab7fe-3b9f-4693-a2af-49c4d20d11d8">